src/main/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptor.java

```diff
@@ -0,0 +1,69 @@
+import uk.gov.companieshouse.api.accounts.AttributeName;
+import uk.gov.companieshouse.api.accounts.LinkType;
+import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
+import uk.gov.companieshouse.api.accounts.service.CompanyAccountService;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@Component
+public class CompanyAccountInterceptor extends HandlerInterceptorAdapter {
+
+    @Autowired
+    private CompanyAccountService companyAccountService;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("company-accounts.api.ch.gov.uk");
+
+    /**
+     * This class extracts the 'company_account' parameter passed via the URI then validates it.
+     * Validation is carried out via a database lookup for a CompanyAccountEntity Object then
+     * matching the retrieved Entities 'self' link to the session stored Transactions
+     * 'company_account' link. Providing this validation passes it assigns the CompanyAccountEntity
+     * to the session.
+     *
+     * @param request - current HTTP request
+     * @param response - current HTTP response
+     * @param handler - chosen handler to execute, for type and/or instance evaluation
+     * @return true if the execution chain should proceed with the next interceptor or the handler
+     * itself.
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
+            Object handler) {
+        Map<String, String> pathVariables = (Map) request
+                .getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+        String companyAccountId = pathVariables.get("companyAccountId");
+        CompanyAccountEntity companyAccountEntity = companyAccountService
+                .findById(companyAccountId);
+        Transaction transaction = (Transaction) request.getSession()
+                .getAttribute(AttributeName.TRANSACTION.getValue());
+        HttpSession session = request.getSession();
+        if (transaction != null && companyAccountEntity != null) {
+            String accountsLink = transaction.getLinks().get(LinkType.COMPANY_ACCOUNT.getLink());
+            String accountsSelf = companyAccountEntity.getData().getLinks().get(LinkType.SELF.getLink());
+            if (accountsLink.equals(accountsSelf)) {
+                session.setAttribute(AttributeName.COMPANY_ACCOUNT.getValue(),
+                        companyAccountEntity);
+                return true;
+            }
+        }
+        StringBuilder sb = new StringBuilder("CompanyAccountInterceptor failed on preHandle");
+        if (transaction != null) {
+            sb.append(" for transaction ").append(transaction.toString());
+        }
+        sb.append(".");
+        LOGGER.error(sb.toString());
+        return false;
+    }
```

Comment:
What does false return to the user?

Reply:
At the moment nothing it's only there to get the code to compile, once the Global error handling is in place I will look to address this possibly throwing an exception that spawns either a 400 or 404 depending what failed in the interceptor.